src/FunctionApps/python/IntakePipeline/__init__.py

```diff
@@ -0,0 +1,27 @@
+import logging
+
+import azure.functions as func
+
+from IntakePipeline.transform import transform_bundle
+from IntakePipeline.linkage import add_patient_identifier
+from IntakePipeline.fhir import read_fhir_bundles, upload_bundle_to_fhir_server
+
+
+def run_pipeline():
+    for bundle in read_fhir_bundles():
+        transform_bundle(bundle)
+        add_patient_identifier(bundle)
+        upload_bundle_to_fhir_server(bundle)
```

Comment:
Again, I like this. Point to note here is that Graham found that the real VA data is broken up into a massive set of smaller files, each around ~20 mb. That's _key_ for us (or at least finding that out was key to me), because it actually solves our memory bottleneck problem. If the FHIR server only converts one bundle (= one file = 20 mb) at a time, then our Function App will never run into that 1.5 gb limit. There will be another step at the top of this `run_pipelin()` function before we `transform`, where we run `convert_bundle_by_message_to_bundle_by_resource`, and we might wind up doing some data returns in between, (e.g.
```
def run_pipeline():
    for bundle in read_fhir_bundles():
        resource_packet = convert_bundle_by_message_to_bundle_by_resource(bundle)
        resource_packet = transform_bundle(resource_packet)
        resource_packet = add_linking_identifier(resource_packet)
        resource_packet = link_duplicate_records(resource_packet)
        upload_bundle_to_fhir_server(resource_packet)
```
), but overall this is exactly what I had in mind.

Reply:
One of the big ideas here is to keep the bundles intact for now - Azure's FHIR bulk import doesn't work, so there's no requirement to group them.

Reply:
For sure, no requirement to regroup them but we do want to do that for the record linkage function. That's how intra-batch linkage finds duplicate patients and that's how we can iterate over non-patients to change the ID they point to. It's just really helpful to have things split up like that, and since Dan already has the function (or 80% of one) hopefully it's a relatively low effort to plug in.

Reply:
I'm curious...  In what way does Azure's bulk import not work?  What has been attempted or what documentation is out there that does not work properly?

Reply:
Jesse got `The bulk import feature is currently available in the open source FHIR server. It's not available in Azure Health Data Services yet.` a couple weeks ago on the $import endpoint